haskell/test/SMT.hs

```diff
@@ -176,45 +175,45 @@ resultToSs sval =
 
 storageToSs :: Storage -> SimpleStorage
 storageToSs storage = SimpleStorage
-  { ssMintingAllowances = sMintingAllowances storage
-  , ssLedger = bmMap $ sLedger storage
-  , ssOwner = rOwner $ sRoles storage
-  , ssMasterMinter = rMasterMinter $ sRoles storage
-  , ssPauser = rPauser $ sRoles storage
-  , ssPendingOwner = rPendingOwner $ sRoles storage
+  { ssMintingAllowances = Map.mapKeys addressToImplicitAddress $ sMintingAllowances storage
+  , ssLedger = Map.mapKeys addressToImplicitAddress $ bmMap $ sLedger storage
+  , ssOwner = addressToImplicitAddress $ rOwner $ sRoles storage
+  , ssMasterMinter = addressToImplicitAddress $ rMasterMinter $ sRoles storage
+  , ssPauser = addressToImplicitAddress $ rPauser $ sRoles storage
+  , ssPendingOwner = addressToImplicitAddress <$> rPendingOwner (sRoles storage)
```

Comment:
I'm not sure if it makes sense, but this bunch of unsafe coercions seems kind of suspicious.

Reply:
I think it's fine here - these tests keep converting between `Storage` and `SimpleStorage`, back and forth. The former contains `Address`es, the latter contains `ImplicitAddress`es.

BTW I added the `unsafe*` prefix here as well.

Reply:
Ah I see your point! Yeah that would be a bigger effort ^^

Frankly, the stablecoin tests could use a deep refactoring. In particular, [these tests](https://github.com/tqtezos/stablecoin/blob/bc167c9d043dbcb0d95c297e0b73fde7cf314ee4/haskell/test/Lorentz/Contracts/Nettest/Nettest.hs) are a remnant from old times.

Before Cleveland was a thing, we used to write the vast majority of our tests via the emulator only, and then have a handful of expensive "nettest" network tests, just as a sort of smoke tests.

But now that _all_ the tests are written with Cleveland and are run on both the emulator and the network, these smoke tests are probably redundant.

Also, we didn't have any way of splitting network tests into individual tasty tests, so you can see [they're still all run as one](https://github.com/tqtezos/stablecoin/blob/bc167c9d043dbcb0d95c297e0b73fde7cf314ee4/haskell/test/Lorentz/Contracts/Nettest/Nettest.hs#L384-L389) :sweat_smile: